### ISSUE TITLE
Tale workspace file/folder operations

### ISSUE THREAD

**Issue by craig-willis** [whole-tale/dashboard#337]
Per discussion from https://github.com/whole-tale/dashboard/issues/304, moving workspace operations to it's own issue.

Goal:
* User can complete supported actions (listed below) in the Workspace folder for a Tale.

Supported actions:
* [x]  Upload files
* [x]  Create folders
* [x]  Rename folder/file
* [x]  Copy files from/to Home
* [x]  Move files from/to Home
* [x]  Remove folder/files
* [x] Select data from other workspaces

Completion criteria:
* [x]  User can manage files/folders using supported actions
* [ ]  Documentations updated
* [x]  Manual test case provided

**Comment by cyberdante** [comment]
### Test Case

1. Checkout this branch locally, pointed at `girder.dev.wholetale.org`: `git fetch --all && git checkout tale-workspace-operations`
2. Start Ember: `ember s`
3. Login to the dev instance that you just started (`http://localhost:4200` most likely)
4. Launch a Tale that you own or create one if needed
5. Select the Tale Instance that you just launched and choose "Run" at the top
    * You should see the "Interact", "Files", and "Metadata" tabs
    * The Interact tab should be selected by default and show the familiar Tale view (e.g. iframe or link)
6. Choose the Files tab
    * You should see a vertical navigation menu with three tabs: "Home", "External Data", and "Tale Workspace".
7. Navigate to the `home` directory and make sure you have some files and folders in there. If not, upload some files and/or create folder(s).
8. Select "Tale Workspace"
    * The first time there should be no data inside this folder. 
9. Use the rounded + button on the right of that view and click on `Select Data...`
    * A new modal should open displaying two tabs on the left: `home` (with your `home` directory content, and `tale workspaces`, which shows all workspaces' folders across all tales
    * Files/folders can be selected/deselected by clicking them, and the user can navigate inside folders as desired and back to the root.
    * Selected files can be copied from either the `home` folder or any tale workspace into the current tale's `workspace`
    * Selected files can be moved from either the `home` folder or any tale workspace into the current tale's `workspace` (there is a know bug related to the moving operation and reported in [#208](https://github.com/whole-tale/girder_wholetale/issues/208)
    * Copy and move some files and folders into the current `workspace`
10. Select the small `chevron-down` icon next to any file or folder to display a context menu and then click on `Copy to Home`.
11. Go to the `home` folder. Check that the files/folders you copied appear in your `home` folder.
12. Go back to the `workspace` directory. Click on the small `chevron-down` icon again but then click on the `Move to...` option.
    * A modal to select where to move your item to should be open. Select the checkbox next to `Home` and then click on the `Move` button.
    * Check that the file/folder moved was removed from your `workspace` and now appears inside the `home` directory.

**Comment by craig-willis** [comment]
@cyberdante where is the PR for this?

**Comment by cyberdante** [comment]
@craig-willis take a look at [this comment](https://wholetale.slack.com/archives/C2F1FQEDC/p1546920302021800) on Slack.

**Comment by craig-willis** [comment]
Thanks, @cyberdante. I didn't see that Slack message (the #dev channel is pretty noisy with Github updates).  For anyone else following along, here's the message from 1/7/2018:
```
I have just pushed the changes to copy/move from the `home` folder to a tale's `workspace` folder. After what we saw today during a big rebase, the branch I have with the work needs also cleaning to keep a healthy history of the commits, but I'd rather have someone check functionality before going through the rebase pain. Would anyone care to check the work in branch `tale-workspace-operations`? There is a known bug involving the moving operation, so keep that in mind
thanks beforehand (edited) 
```

Yes, we'll take a look.  Go ahead and create a PR -- feel free to put "WIP" (work in progress) in the title to indicate that it's not complete -- and indicate that it is related to this issue (i.e., put "Fixes #337" in the description).   We can also use the PR to provide rebase feedback.
